analysis/src/block_enforcer.rs

```diff
@@ -0,0 +1,37 @@
+//! For all machines, enforce that the `function_id` can only change when the `latch` is on
+
+use ast::asm_analysis::{AnalysisASMFile, PilBlock};
+use number::FieldElement;
+
+use crate::utils::parse_pil_statement;
+
+pub fn enforce<T: FieldElement>(mut file: AnalysisASMFile<T>) -> AnalysisASMFile<T> {
+    for machine in file.machines.values_mut() {
+        let function_id = machine.function_id.as_ref().unwrap();
+        let latch = machine.latch.as_ref().unwrap();
+        let first_step = "_first_step";
```

Comment:
so every machine has their own first step column?

Reply:
Yes! They are different AIRs in the end.